src/test/path-transformers_test.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/// <reference path="../../node_modules/@types/mocha/index.d.ts" />
+
+import {assert} from 'chai';
+import {join as pathJoin, sep as pathSeparator} from 'path';
+
+import {isPlatformWindows, pathFromUrl, urlFromPath} from '../path-transformers';
+
+const WindowsRootPath = 'C:\\Users\\TEST_USER\\TEST_ROOT';
+const MacRootPath = '/Users/TEST_USER/TEST_ROOT';
+const RootPath = isPlatformWindows() ? WindowsRootPath : MacRootPath;
+
+suite('pathFromUrl()', () => {
+
+  test('creates a filesystem path using the platform separators', () => {
+    const otherSeparator = pathSeparator === '/' ? '\\' : '/';
+    const path = pathFromUrl(RootPath, '/some/url/pathname');
+    assert.include(path, pathSeparator);
+    assert.notInclude(path, otherSeparator);
+  });
+
+  test('returns a path if url is absolute', () => {
+    const path = pathFromUrl(RootPath, '/absolute/path');
+    assert.equal(path, pathJoin(RootPath, 'absolute', 'path'));
```

Comment:
So there are a few places in Build where we use `pathFromUrl` to generate the filesystem path where something will be written to, because right now the Analyzer doesn't have a symmetrical parse/production interface for paths<->urls; urls are resolved and loaded by Analyzer and there's no means by which Analyzer can be given a url it's never seen, for example, and decide what the filesystem path should be.

On your example, I need to understand if you're saying there is a legitimate case for `../paper-button/paper-button.html` to ever become `rootDir/bower_components/paper-button/paper-button.html` when the base url is not `rootDir/bower_components/{anything}` in a build scenario.